polus-csv-statistics-plugin/src/main.py

```diff
@@ -0,0 +1,217 @@
+        out = str(Path(outDir).joinpath(feat_file.name).absolute())
+        with open(fpath,'r') as fr:
+            with open(str(out),'w') as fw:
+                # Read the first line, which should contain headers
+                first_line = fr.readline()
+                headers = first_line.rstrip('\n').split(',')
+                var_ind = {key:val for key,val in enumerate(headers)} # map headers to line positions
+                
+                # If no column is labeled file, throw an error
+                if 'file' not in headers:
+                    ValueError('At least one column must have a header title file.')
+                
+                # Generate the output dictionary template and format string
+                line_dict = {'file': 'NaN'}
+                for key in headers:
+                    if key=='file':
+                        continue
+                    for stat in statistics:
+                        line_dict[key + '_' + stat] = 'NaN'
+                        # Generate the line template
+                line_template = ','.join([('{' + h + '}') for h in line_dict.keys()]) + '\n'
+
+                # Write headers to the new file
+                fw.write(','.join(line_dict.keys()) + '\n')
+
+                # Get the first line of data
+                line = fr.readline()
+
+                # Read each line in the stitching vector
+                fnum = 0
+                fcheck = 0
+                while line:
+                    # Parse the current line as a dictionary
+                    p_line = {var_ind[ind]:val for ind,val in enumerate(line.rstrip('\n').split(','))}
+                    for key,val in p_line.items():
+                        v = get_number(val)
+                        p_line[key] = [v]
+
+                    # Loop through rows until the filename changes
+                    line = fr.readline()
+                    np_line = {var_ind[ind]:val for ind,val in enumerate(line.rstrip('\n').split(','))}
+                    while line and p_line['file'] == np_line['file']:
+                        # Store the values in a feature list
+                        for key,val in np_line.items():
+                            if isinstance(val,list):
+                                p_line[key].append(get_number(val))
+
+                        # Get the next line
+                        line = fr.readline()
+                        np_line = {var_ind[ind]:val for ind,val in enumerate(line.rstrip('\n').split(','))}
+
+                    # Get the mean of the feature list, save in the file dictionary
+                    for key,val in p_line.items():
+                        # Set the file name
+                        if key=='file':
+                            line_dict['file'] = val[0]
+                            continue
+                        
+                        # Grab only float values
+                        inp_data = [d for d in val if isinstance(d,float)]
```

Comment:
Wouldn't you also want to grab Ints? I was looking earlier and it doesn't seem like you do a conversion to all as floats. For things like "neighbors" or "Euler number" they'll be Ints right?

Reply:
Python supports mixed arithmetic .So  the number with the narrower type is widened to that of the other when a operation is performed. For e.g.  1 +3.0 becomes 4.0 and not 4.